File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.java
class Solution {
    public int solution(int n, int[] stations, int w) {
        		
        int answer = 0;
        int range = w * 2 + 1;

		int size = 0;
		int rightIndex = 0;

		for (int i = 0; i < stations.length; i++) {
			int target = stations[i];

			size = target - rightIndex - 1 - w;
			rightIndex = target + w;

            if(size <= 0) continue;

			int res = size % range != 0 ? size / range + 1 : size / range;
			answer += res;
		}

		if (rightIndex <= n) {
			size = n - rightIndex;
			int res = size % range != 0 ? size / range + 1 : size / range;
			answer += res;
		}
        return answer;
    }
}

File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.js
function solution(n, stations, w) {
    let answer = 0;
    let pos = 1;
    let index = 0;
    
    while (pos <= n) {
        if (pos >= stations[index] - w && pos <= stations[index] + w) {
            pos = stations[index++] + w;
        } else {
            pos += 2 * w;
            answer++;
        }
        pos++;
    }
    
    return answer;
}